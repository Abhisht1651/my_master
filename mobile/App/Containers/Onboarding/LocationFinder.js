// @flow
import { NavigationContainerProps } from '@react-navigation/native'
import React from 'react'
import { Text, Image, View, TouchableOpacity, Platform } from 'react-native'
import CurrentLocationIcon from '../../Images/Icons/icon-currentlocation.svg'
import styles from '../Styles/Onboarding/LocationFinderStyles'
import Images from '../../Themes/Images'
// Styles
type Props = NavigationContainerProps & {}

const LocationFinder = (props: Props) => {
  return (
    <View style={styles.centeredView}>
      {Platform.OS === 'ios' && <View style={styles.statusBar} />}
      <View style={styles.header}>
        <TouchableOpacity
          onPress={() => props.navigation.goBack()}
          style={styles.backArrowContainer}>
          <Image source={Images.leftArrow} style={styles.backArrow} />
        </TouchableOpacity>

        <TouchableOpacity onPress={() => props.navigation.goBack()}>
          <View>
            <Image source={Images.closeIconPurple} style={styles.cross} />
          </View>
        </TouchableOpacity>
      </View>

      <Text style={styles.modalText}>Looking For Your Current Location</Text>
      <CurrentLocationIcon />
    </View>
  )
}
export default LocationFinder
