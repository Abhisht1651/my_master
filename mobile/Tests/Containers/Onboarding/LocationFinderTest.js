import { Image } from 'react-native'
import React from 'react'
import { shallow } from 'enzyme'
import LocationFinder from '../../../App/Containers/Onboarding/LocationFinder'
import CurrentLocationIcon from '../../../App/Images/Icons/icon-currentlocation.svg'

const createTestProps = () => ({
  navigation: {
    navigate: jest.fn()
  }
})

describe('LocationFinder', () => {
  describe('rendering', () => {
    let wrapper
    let props // use type "any" to opt-out of type-checking
    beforeEach(() => {
      props = createTestProps({})
      wrapper = shallow(<LocationFinder props />) // no compile-time error
      let x = props
      if (x) x = props
    })
    it('should match to snapshot', () => {
      expect(wrapper).toMatchSnapshot()
    })
    it('should render 2 <Image />', () => {
      expect(wrapper.find(Image)).toHaveLength(2)
    })
    it('should render 1 <CurrentLocationIcon />', () => {
      expect(wrapper.find(CurrentLocationIcon)).toHaveLength(1)
    })
  })
})
