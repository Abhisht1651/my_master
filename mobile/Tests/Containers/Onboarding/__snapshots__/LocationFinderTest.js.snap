// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocationFinder rendering should match to snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "white",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "white",
        "height": 58.188802083333336,
        "width": "100%",
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "height": 100.74479166666667,
        "justifyContent": "space-between",
        "paddingHorizontal": 50,
        "width": "100%",
      }
    }
  >
    <ForwardRef
      onPress={[Function]}
    >
      <Image
        source={
          Object {
            "default": "",
          }
        }
        style={
          Object {
            "height": 31.265625,
            "resizeMode": "contain",
            "width": 36,
          }
        }
      />
    </ForwardRef>
    <ForwardRef
      onPress={[Function]}
    >
      <View>
        <Image
          source={
            Object {
              "default": "",
            }
          }
          style={
            Object {
              "height": 61.497400000000006,
              "resizeMode": "contain",
              "tintColor": "#7B46E4",
              "width": 34.575,
            }
          }
        />
      </View>
    </ForwardRef>
  </View>
  <Text
    style={
      Object {
        "color": "#1F1F60",
        "fontFamily": "BananaGrotesk-Medium",
        "fontSize": 32,
        "letterSpacing": -0.11,
        "marginBottom": 255.3359375,
        "marginHorizontal": 180,
        "marginTop": 138.95833333333334,
        "textAlign": "center",
      }
    }
  >
    Looking For Your Current Location
  </Text>
  <SvgMock />
</View>
`;
